search-card.js

```diff
@@ -87,7 +87,10 @@ class SearchCard extends LitElement {
 
     for (var entity_id in this.hass.states) {
       if (entity_id.indexOf(searchText) >= 0) {
-        this.data.push(entity_id);
+        this.data.indexOf(entity_id) === -1 ? this.data.push(entity_id) : null;
```

Comment:
I'd suggest placing all of the if statements into a single if since we want to add anything anyways so:
```
if (
(entity_id.indexOf(searchText) >= 0) || /* If statement 1 \*
(this.hass.states[entity_id].attributes.friendly_name.search(new RegExp(searchText, 'i')) >= 0) /* if statement two */
) {
```

Other than that it looks good to me.